register_buildpack_test.go

```diff
@@ -0,0 +1,130 @@
+			h.AssertNil(t, fakeAppImage.SetLabel("io.buildpacks.buildpackage.metadata",
+				`{"id":"heroku/java-function","version":"1.1.1","stacks":[{"id":"heroku-18"},{"id":"io.buildpacks.stacks.bionic"},{"id":"org.cloudfoundry.stacks.cflinuxfs3"}]}`))
+			fakeImageFetcher.RemoteImages["buildpack/image"] = fakeAppImage
+
+			fakeLogger := logging.NewLogWithWriters(&out, &out)
+			subject = &Client{
+				logger:       fakeLogger,
+				imageFetcher: fakeImageFetcher,
+			}
+		})
+
+		it.After(func() {
+			_ = fakeAppImage.Cleanup()
+		})
+
+		it("should return error for an invalid image", func() {
+			fakeAppImage = fakes.NewImage("invalid/image", "", &fakeIdentifier{name: "buildpack-image"})
+			h.AssertNil(t, fakeAppImage.SetLabel("io.buildpacks.buildpackage.metadata", `{}`))
+
+			h.AssertNotNil(t, subject.RegisterBuildpack(context.TODO(),
+				RegisterBuildpackOptions{
+					ImageName: "invalid/image",
+					Type:      "github",
+					URL:       "https://github.com/jkutner/buildpack-registry",
+				}))
+		})
+
+		it("should return error for missing image label", func() {
+			fakeAppImage = fakes.NewImage("missinglabel/image", "", &fakeIdentifier{name: "buildpack-image"})
+			h.AssertNil(t, fakeAppImage.SetLabel("io.buildpacks.buildpackage.metadata", `{}`))
+			fakeImageFetcher.RemoteImages["missinglabel/image"] = fakeAppImage
+
+			h.AssertNotNil(t, subject.RegisterBuildpack(context.TODO(),
+				RegisterBuildpackOptions{
+					ImageName: "missinglabel/image",
+					Type:      "github",
+					URL:       "https://github.com/jkutner/buildpack-registry",
+				}))
+		})
+
+		when("registry type is github", func() {
+			it("should open a github issue in the browser", func() {
+				const expectedURL = `https://github.com/jkutner/buildpack-registry/issues/new?body=%0A%23%23%23+Data%0A%0A%60%60%60toml%0Aid+%3D+%22heroku%2Fjava-function%22%0Aversion+%3D+%221.1.1%22%0Aaddr+%3D+%22buildpack-image%22%0A%60%60%60&title=ADD+heroku%2Fjava-function%401.1.1`
+
+				switch runtime.GOOS {
+				case "linux":
+					execCommand = getFakeExecCommand(t, "xdg-open", expectedURL)
+				case "windows":
+					execCommand = getFakeExecCommand(t, "rundll32", "url.dll,FileProtocolHandler", expectedURL)
+				case "darwin":
+					execCommand = getFakeExecCommand(t, "open", expectedURL)
+				default:
+					// do nothing
+				}
+
+				h.AssertNil(t, subject.RegisterBuildpack(context.TODO(),
+					RegisterBuildpackOptions{
+						ImageName: "buildpack/image",
+						Type:      "github",
+						URL:       "https://github.com/jkutner/buildpack-registry",
+					}))
+			})
+
+			it("should throw error if missing URL", func() {
+				h.AssertError(t, subject.RegisterBuildpack(context.TODO(),
+					RegisterBuildpackOptions{
+						ImageName: "buildpack/image",
+						Type:      "github",
+						URL:       "",
+					}), "missing github URL")
+			})
+		})
+	})
+}
+
+func getFakeExecCommand(t *testing.T, expectedCmd string, expectedArgs ...string) func(command string, args ...string) *exec.Cmd {
```

Comment:
A few points here:

1. Terminology here made it confusing for me to understand what was going on. This isn't a "fake" in the way I would understand a fake to work. It's really more of a "mock" with expectations/asserting. Maybe changing the name to something like `createExpectedCommand`.
2. I wonder if we can make it more architecturally sound if this were to be moved to the `registry` package (as commented [here](https://github.com/buildpacks/pack/pull/716/files#diff-f4dc727bc4e1e13f3f5bcc35f4da913aR103)) and then instead of relying on a package variable `execCommand` we used an interface that we can properly mock. eg. `BrowserExecFactory`.